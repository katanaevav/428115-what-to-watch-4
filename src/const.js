export const START_URL = `https://4.react.pages.academy`;

export const NO_FILTER = `All genres`;

export const DELAY_BEFORE_START_PREVIEW = 1000;
export const MAX_RENDERED_MOVIES_AT_TIME = 8;

export const MAX_SIMILAR_MOVIES_COUNT = 4;

export const REVIEWS_COLUMNS_COUNT = 2;

export const MIN_VOLUME = 0.0;
export const NO_FULLSCREEN = false;

export const TABS_NAMES = [`Overview`, `Details`, `Reviews`];

export const Screens = {
  MAIN_SCREEN: 0,
  MOVIE_PAGE_SCREEN: 1,
  CINEMA_SCREEN: 2,
  AUTH_SCREEN: 3,
  ADD_REVIEW_SCREEN: 4,
};

export const SavingStatus = {
  SUCCESS: `SUCCESS`,
  FAIL: `FAIL`,
};

export const Tabs = {
  OVERVIEW_TAB: 0,
  DETAILS_TAB: 1,
  REVIEWS_TAB: 2,
};

export const MONTH_NAMES = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
