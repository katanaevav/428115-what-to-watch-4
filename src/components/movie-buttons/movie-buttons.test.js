import React from "react";
import renderer from "react-test-renderer";
import MovieButtons from "./movie-buttons.jsx";
import {AuthorizationStatus} from "../../const.js";
import {Router} from "react-router-dom";
import history from "../../history.js";

it(`Should MovieButtons component render correctly`, () => {
  const tree = renderer.create(
      <Router history= {history}>
        <MovieButtons
          isMainScreen = {false}
          movieId = {1}
          authorizationStatus = {AuthorizationStatus.AUTH}
          isFavorite = {false}
          onFavoriteButtonClick = {() => {}}
        />
      </Router>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should MovieButtons component render correctly for main screen`, () => {
  const tree = renderer.create(
      <Router history= {history}>
        <MovieButtons
          isMainScreen = {true}
          movieId = {1}
          authorizationStatus = {AuthorizationStatus.AUTH}
          isFavorite = {false}
          onFavoriteButtonClick = {() => {}}
        />
      </Router>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
