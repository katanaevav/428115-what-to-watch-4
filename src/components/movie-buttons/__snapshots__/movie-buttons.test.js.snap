// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieButtons component render correctly 1`] = `
<div
  className="movie-card__buttons"
>
  <a
    className="btn btn--play movie-card__button"
    href="/player/1"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="/sprite.svg#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </a>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlinkHref="/sprite.svg#add"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
  <a
    className="btn movie-card__button"
    href="/films/1/review"
    onClick={[Function]}
  >
    Add review
  </a>
</div>
`;

exports[`Should MovieButtons component render correctly for main screen 1`] = `
<div
  className="movie-card__buttons"
>
  <a
    className="btn btn--play movie-card__button"
    href="/player/1"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="/sprite.svg#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </a>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlinkHref="/sprite.svg#add"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
  
</div>
`;
